yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -809,7 +809,13 @@ object Client extends Logging {
       }
     }
     addFileToClasspath(new URI(sparkJar(sparkConf)), SPARK_JAR, env)
-    populateHadoopClasspath(conf, env)
+    // Since we have a spark assembly that is including all the yarn and other dependencies we need,
```

Comment:
"Since" makes it seem like we know this is the case.  I'd say "Because the Spark assembly may already include Hadoop and its dependencies..."